packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -4487,6 +4780,203 @@ describe('TransactionController', () => {
+                op: 'remove',
+                path: ['networkConfigurations', 'goerli'],
+                value: 'foo',
+              },
+              {
+                op: 'remove',
+                path: ['networkConfigurations', 'customNetworkClientId-1'],
+                value: 'foo',
+              },
+            ]);
+          }, 0);
+        },
+      );
+
+      newController({
+        options: {
+          messenger: mockMessenger.messenger,
+          getNetworkClientRegistry: mockGetNetworkClientRegistry,
+          enableMultichain: true,
+          hub,
+        },
+      });
+      await receivedEvents;
+    });
+  });
+  it('should handle additions to the networkController registry', async () => {
+    const hub = new EventEmitter() as TransactionControllerEventEmitter;
+    const mockGetNetworkClientRegistry = jest.fn();
+    mockGetNetworkClientRegistry.mockImplementationOnce(() => ({
+      sepolia: {
+        configuration: {
+          chainId: ChainId.sepolia,
+        },
+      },
+    }));
+
+    hub.on('tracking-map-init', () => {
+      mockGetNetworkClientRegistry.mockClear();
+      mockGetNetworkClientRegistry.mockImplementation(() => ({
+        sepolia: {
+          configuration: {
+            chainId: ChainId.sepolia,
+          },
+        },
+        goerli: {
+          configuration: {
+            chainId: ChainId.goerli,
+          },
+        },
+      }));
+    });
+
+    const receivedEvents = new Promise((resolve) => {
+      let expectedNetworkClientIds = ['goerli', 'sepolia'];
+      hub.on('tracking-map-add', (networkClientId) => {
+        expect(expectedNetworkClientIds).toContain(networkClientId);
+        expectedNetworkClientIds = expectedNetworkClientIds.filter(
+          (v) => v !== networkClientId,
+        );
+        if (expectedNetworkClientIds.length === 0) {
+          resolve(undefined);
+        }
+      });
+    });
+    const mockMessenger = buildMockMessenger({});
+    (mockMessenger.messenger.subscribe as jest.Mock).mockImplementation(
+      (_type, handler) => {
+        setTimeout(() => {
+          handler({}, [
+            {
+              op: 'add',
+              path: ['networkConfigurations', 'goerli'],
+              value: 'foo',
+            },
+          ]);
+        }, 0);
+      },
+    );
+
+    newController({
+      options: {
+        messenger: mockMessenger.messenger,
+        getNetworkClientRegistry: mockGetNetworkClientRegistry,
+        enableMultichain: true,
+        hub,
+      },
+    });
+    await receivedEvents;
+  });
+
+  describe('startIncomingTransactionPolling', () => {
+    it('should start the global incoming transaction helper when no networkClientIds provided', () => {
+      const controller = newController({
+        options: {
+          enableMultichain: true,
+        },
+      });
+
+      controller.startIncomingTransactionPolling([]);
+
+      expect(incomingTransactionHelperMocks[0].start).toHaveBeenCalledTimes(1);
```

Comment:
TODO: unclear to me whether we need these tests and should rewrite them from a more functional perspective.

Reply:
The `IncomingTransactionHelper` is a distinct unit in the wider system of the `TransactionController` and so already has a full suite of its own unit tests.

The goal of any unit test should be isolating the specific logic you want to test, the desired result of this method is that it should call a separate tested unit.

If we test the side effects here, then we're just duplicating the assertions within the incoming transaction helper tests, testing two units rather than one, plus adding complexity to this unit test.

Reply:
These tests were really more appropriately added to the new Integration test suite since they are specifically aiming to ensure that the correct instance of the incomingTransactionHelper is used in this particular scenario. Removed this test in [this PR](https://github.com/MetaMask/core/pull/3885/files#diff-f451a8bcb8e653dc79707c0e429c7f3c57e5b64e7e6b749a22062a01496a506cL4957). Removing a remaining test of this sort (`should update the global incoming transactions when no networkClientIds provided`) and replacing with one that is more functionally oriented: `should update the incoming transactions for the globally selected network when no networkClientIds provided` (to `updateIncomingTransactions`)